### PR TITLE
return nil eather than 0 if total is zero

### PR DESCRIPTION
**Pull request by willmcb**
## Description
Removes zero from % columns in reports when there are no values present.

## Self-review checklist

* [ ] (1) Quick stakeholder demo done OR
* [ ] (2) ...bug with before and after screenshots
* [ ] (3) Tests passing
* [ ] (4) Branch ready to be merged (not work in progress)
* [ ] (5) No superfluous changes in diff
* [ ] (6) No TODO's without new ticket numbers
* [ ] (7) PR Prefixed with ticket number e.g. `CT-7654 ...`

### Related JIRA tickets
[CT-2453](https://dsdmoj.atlassian.net/browse/CT-2453)

### CODE REVIEW

File: app/services/stats/calculations.rb [hunk 1]
```diff
@@ -128,7 +128,7 @@ def self.calculate_bu(row)
 
     def self.calculate_percentage(value, total)
       if total == 0
-        0.0
+        nil
```
Comment (ymao2) [inline app/services/stats/calculations.rb:131]: @willmcb,  changing here seems working with xlsx,   initially, I was thinking of not changing here as calculations.rb is not only used by this report, but for monthly performance report too (from another side, it maybe a benefit),  and also it is sort of arguable thing for returning nil rather 0 for math calculation 🤔 

another way for this ticket is to change the function `to_csv` in `base_business_unit_performance_report.rb`  between line 102 and 105,  let me know what you think, sorry my suggestion maybe is too late 😳

Comment (willmcb) [reply app/services/stats/calculations.rb:131]: I have no strong opinions - only really that if this is okay and _works_ I'd rather just merge it 👍

File: spec/features/admin/business_units_spec.rb [hunk 2]
```diff
@@ -39,7 +39,7 @@
 
     teams_areas_page.add_area_field.set 'This is another area'
     teams_areas_page.add_button.click
-    teams_areas_page.wait_until_existing_areas_visible count: 2
+    sleep(2)
```
Comment (ymao2) [inline spec/features/admin/business_units_spec.rb:42]: IMO not remove the line and move the sleep(2) above this line, the count of 2 means 2 elements not 2 seconds?

**Review by ymao2 — APPROVED** [review]
Happy to approve 🤓 ! only just some minor comments which shouldn't hold this PR

File: spec/features/admin/business_units_spec.rb (hunk 2 above)
Comment (willmcb) [reply spec/features/admin/business_units_spec.rb:42]: Yes I think you're correct - though as it is not an expectation and it passes with the sleep I think it is okay.

**Pull request merged**